llama.cpp

```diff
@@ -3713,6 +3839,44 @@ struct llama_model_loader {
         std::vector<no_init<uint8_t>> read_buf;
         std::vector<std::future<std::pair<ggml_tensor *, bool>>> validation_result;
 
+#if defined(GGML_USE_CUDA)
+		// 4 staging buffers for async uploads, each sized 1MB seems to be a good default for single NVMe drives.
+		// NVMe raid configurations might require more / larger buffers.
+        constexpr size_t num_buffers = 4;
+        constexpr size_t buffer_size = 1 * 1024 * 1024; // 1MB
+
+        std::vector<ggml_backend_buffer_t> host_buffers;
+        std::vector<void*> host_ptrs;
+        std::vector<ggml_backend_event_t> events;
+        size_t buffer_idx = 0; // buffer to use for async loads
+
+        ggml_backend_t cuda_backend = nullptr;
+        if (!use_mmap) {
```

Comment:
Since this is not compatible with tensor validation, it should also be disabled when it is enabled.

```suggestion
        if (!use_mmap && !check_tensors) {
```